### PR TITLE
Added Braze Actions

### PR DESCRIPTION
**Pull request by zzhaobraze**
Added Looker Action to send data to Braze via REST API.

### CODE REVIEW

File: src/actions/braze/braze.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
+import * as req from "request-promise-native"
+import * as url from "url"
+
+import * as Hub from "../../hub"
+
+export enum BrazeConfig {
```
Comment (wilg) [inline src/actions/braze/braze.ts]: Doesn't look like this needs to be exported.

File: src/actions/braze/braze.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+  name = "braze"
+  label = "Braze"
+  description = "Ensure there's a '" + BrazeConfig.BRAZE_ID_TAG + "' field tagged in a results."
+    + " MAX EXPORT: " + BrazeConfig.MAX_EXPORT + "."
+  iconName = "braze/braze.svg"
+  supportedActionTypes = [Hub.ActionType.Query]
+  supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
+  supportedFormattings = [Hub.ActionFormatting.Unformatted]
+  // supportedFormats = [Hub.ActionFormat.JsonDetail]
```
Comment (wilg) [inline src/actions/braze/braze.ts]: Please remove commented out code.

File: src/actions/braze/braze.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      description: "Name of export (Appends to Custom Attribute Array '" +
+        BrazeConfig.LOOKER_ATTRIBUTE_NAME + "'). Defaults to '" + BrazeConfig.EXPORT_DEFAULT_VALUE + "'." ,
+      required: true,
+      type: "string",
+    },
+    ]
+    return form
+  }
+
+  async sendChunk(urlendpoint: any, apiKey: any, chunk: any[]) {
```
Comment (wilg) [inline src/actions/braze/braze.ts]: Can we add concrete types here instead of `any` throughout the class?

File: src/actions/braze/braze.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        {name: "external_id", label: "external_id"},
+        {name: "braze_id", label: "braze_id"},
+      ],
+      type: "select",
+      default: "external_id",
+    }, {
+      label: "Export Label",
+      name: "braze_segment",
+      description: "Name of export (Appends to Custom Attribute Array '" +
+        BrazeConfig.LOOKER_ATTRIBUTE_NAME + "'). Defaults to '" + BrazeConfig.EXPORT_DEFAULT_VALUE + "'." ,
```
Comment (wilg) [inline src/actions/braze/braze.ts]: You should be able to provide a default value by specifying `default` for a form field.

File: src/actions/braze/test_braze.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import * as chai from "chai"
+// import * as sinon from "sinon"
+// import concatStream = require("concat-stream")
```
Comment (wilg) [inline src/actions/braze/test_braze.ts]: Please remove commented code.

File: src/actions/braze/test_braze.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    }]
+    return form
+  }
+}
+
+const action = new BrazeActionTest()
+
+describe(`${action.constructor.name} unit tests`, () => {
+  describe("action", () => {
+    it("errors if there is no configuration for query", () => {
```
Comment (wilg) [inline src/actions/braze/test_braze.ts]: I think we should have a test case for a successful action.

File: src/actions/braze/braze.ts (hunk 1 above)
Comment (zzhaobraze) [reply src/actions/braze/braze.ts]: fixed thanks

File: src/actions/braze/braze.ts (hunk 3 above)
Comment (zzhaobraze) [reply src/actions/braze/braze.ts]: remove. thanks

File: src/actions/braze/braze.ts (hunk 4 above)
Comment (zzhaobraze) [reply src/actions/braze/braze.ts]: removed thanks

File: src/actions/braze/braze.ts (hunk 2 above)
Comment (zzhaobraze) [reply src/actions/braze/braze.ts]: removed thanks

File: src/actions/braze/braze.ts (hunk 4 above)
Comment (zzhaobraze) [reply src/actions/braze/braze.ts]: added thanks

File: src/actions/braze/test_braze.ts (hunk 6 above)
Comment (zzhaobraze) [reply src/actions/braze/test_braze.ts]: added thanks

File: src/actions/braze/test_braze.ts (hunk 5 above)
Comment (zzhaobraze) [reply src/actions/braze/test_braze.ts]: removed thanks

**Comment by zzhaobraze** [comment]
Thanks for the review @wilg I believe I made all the changes. Please re-review when you get a chance and thanks for all your help.

File: src/actions/braze/braze.ts [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    if (!(request.formParams.braze_key)) {
+      throw "Missing primary Braze key."
+    }
+
+    const exportValue = request.formParams.braze_segment  || String(BrazeConfig.EXPORT_DEFAULT_VALUE)
+    const brazeAttribute = { add : [ exportValue ] }
+    const brazeApiKey = String(request.params.braze_api_key)
+    let totalCnt = 0
```
Comment (wilg) [inline src/actions/braze/braze.ts]: Quick thing: let's rename this `totalCount` please.

Comment (zzhaobraze) [reply src/actions/braze/braze.ts]: renamed. Thanks!

**Pull request merged**